frontend/src/components/ProjectServiceAccountRow.vue

```diff
@@ -5,164 +5,132 @@ SPDX-License-Identifier: Apache-2.0
-        <v-list-item-subtitle>
-          {{username}}
-        </v-list-item-subtitle>
-      </v-list-item-content>
-      <v-list-item-action class="ml-1">
-        <service-account-roles :role-display-names="roleDisplayNames"></service-account-roles>
-      </v-list-item-action>
-      <v-list-item-action v-if="isServiceAccountFromCurrentNamespace && canGetSecrets" class="ml-1">
-        <v-tooltip top>
-          <template v-slot:activator="{ on }">
-            <v-btn v-on="on" icon @click.native.stop="onDownload">
-              <v-icon>mdi-download</v-icon>
-            </v-btn>
-          </template>
-          <span>Download Kubeconfig</span>
-        </v-tooltip>
-      </v-list-item-action>
-      <v-list-item-action v-if="isServiceAccountFromCurrentNamespace && canGetSecrets" class="ml-1">
-        <v-tooltip top>
-          <template v-slot:activator="{ on }">
-            <v-btn v-on="on" icon @click="onKubeconfig">
-              <v-icon>mdi-eye</v-icon>
-            </v-btn>
-          </template>
-          <span>Show Kubeconfig</span>
-        </v-tooltip>
-      </v-list-item-action>
-      <v-list-item-action v-if="canManageServiceAccountMembers" class="ml-1">
-        <v-tooltip top>
-          <template v-slot:activator="{ on }">
-            <v-btn v-on="on" icon @click.native.stop="onEdit">
-              <v-icon>mdi-pencil</v-icon>
-            </v-btn>
-          </template>
-          <span>Change Service Account Roles</span>
-        </v-tooltip>
-      </v-list-item-action>
-      <v-list-item-action v-if="canManageServiceAccountMembers" class="ml-1">
-        <v-tooltip top>
-          <template v-slot:activator="{ on }">
-            <v-btn v-on="on" icon color="red" @click.native.stop="onDelete">
-              <v-icon v-if="isServiceAccountFromCurrentNamespace">mdi-delete</v-icon>
-              <v-icon v-else>mdi-close</v-icon>
-            </v-btn>
-          </template>
-          <span v-if="isServiceAccountFromCurrentNamespace">Delete Service Account</span>
-          <span v-else>Remove Foreign Service Account from Project</span>
-        </v-tooltip>
-      </v-list-item-action>
-    </v-list-item>
-  </div>
+        </div>
+        <div v-if="isServiceAccountFromCurrentNamespace && canDeleteSecrets" class="ml-1">
+          <v-tooltip top>
+            <template v-slot:activator="{ on }">
+              <v-btn v-on="on" icon @click="onRotateSecret">
+                <v-icon>mdi-refresh</v-icon>
+              </v-btn>
+            </template>
+            <span>Rotate Service Account Secret</span>
+          </v-tooltip>
+        </div>
+        <div v-if="canManageServiceAccountMembers" class="ml-1">
+          <v-tooltip top>
+            <template v-slot:activator="{ on }">
+              <v-btn v-on="on" icon @click.native.stop="onEdit">
+                <v-icon>mdi-pencil</v-icon>
+              </v-btn>
+            </template>
+            <span>Change Service Account</span>
```

Comment:
When I change the role of a service account and press `Update` it takes very long until the dialog closes. Did anything change here? At first I thought it didn't react on my button press but then after ~3 seconds the dialog closed

Reply:
I did not change the way this works. Probably some network / server load issue. But there is certainly room for improvement when handling these async actions on the UI, e.g. disabling the button, adding a spinner...
Probably not scope of this PR though